shop-guide/src/operations.ts

```diff
@@ -7,53 +7,53 @@ export const session_topic = "payment_session_id";
-    const handle = await ctxt.invoke(Shop, key).paymentWorkflow();
-
-    // This will block until the payment session is ready
+    // A workflow handle is immediately returned. The workflow continues in the background.
+    const handle = await ctxt.invoke(Shop, key).checkoutWorkflow();
+    ctxt.logger.info(`Checkout workflow started with UUID: ${handle.getWorkflowUUID()}`);
+  
+    // Wait until the payment session is ready
     const session_id = await ctxt.getEvent<string>(handle.getWorkflowUUID(), session_topic);
     if (session_id === null) {
       ctxt.logger.error("workflow failed");
       return "";
     }
-
+  
     return generatePaymentUrls(ctxt, handle.getWorkflowUUID(), session_id);
   }
-
+  
   @Workflow()
-  static async paymentWorkflow(ctxt: WorkflowContext): Promise<void> {
+  static async checkoutWorkflow(ctxt: WorkflowContext): Promise<void> {
     // Attempt to update the inventory. Signal the handler if it fails.
     try {
-      await ctxt.invoke(ShopUtilities).subtractInventory();
+      await ctxt.invoke(ShopUtilities).reserveInventory();
     } catch (error) {
       ctxt.logger.error("Failed to update inventory");
       await ctxt.setEvent(session_topic, null);
       return;
     }
-
+  
     // Attempt to start a payment session. If it fails, restore inventory state and signal the handler.
     const paymentSession = await ctxt.invoke(ShopUtilities).createPaymentSession();
     if (!paymentSession.url) {
       ctxt.logger.error("Failed to create payment session");
-      await ctxt.invoke(ShopUtilities).undoSubtractInventory();
+      await ctxt.invoke(ShopUtilities).undoReserveInventory();
       await ctxt.setEvent(session_topic, null);
       return;
     }
-
-    // Signal the handler with the payment session ID.
+  
+    // Notify the handler of the payment session ID.
     await ctxt.setEvent(session_topic, paymentSession.session_id);
-
-    // Wait for a notification from the payment service.
-    const notification = await ctxt.recv<string>(payment_complete_topic, 30);
-
+  
+    // Await a notification from the payment service.
+    const notification = await ctxt.recv<string>(payment_complete_topic);
+  
     if (notification && notification === 'paid') {
-      // If the payment succeeds, fulfill the order (code omitted for clarity.)
-      ctxt.logger.info(`Payment notification received`);
+      // If the payment succeeds, fulfill the order (code omitted for brevity.)
+      ctxt.logger.info(`Checkout with UUID ${ctxt.workflowUUID} succeeded!`);
     } else {
-      // Otherwise, either the payment failed or timed out.
-      // Code to check the latest session status with the payment service omitted for clarity.
-      ctxt.logger.warn(`Payment failed or timed out`);
-      await ctxt.invoke(ShopUtilities).undoSubtractInventory();
+      // If the payment fails or times out, cancel the order and return inventory.
```

Comment:
We should find a way to keep saying that in a real application you might need to 2x check with the service provider. (Or people could what Harry says about the google demo app? :)

For example:
`// Check session status with payment provider omitted for brevity.`